### PR TITLE
(closes #1404) bug fix for copying of loop variable

### LINKED ISSUE #1404: [psyir] ScopingNode refinecopy() misses loop variable
In #1147 I discovered that the ScopingNode._refinecopy() method fails to correctly copy over the variable associated with a loop. This is a fix for that bug.

### PR DESCRIPTION
**Pull request by arporter**

### CODE REVIEW

**Comment by arporter** [comment]
Another small PR that just fixes the copying of Loop nodes. One for @sergisiso I think.

**Review by sergisiso — CHANGES_REQUESTED** [review]
@arporter The PR looks good. It is a temporal fix until the loop variable is part of the tree, there are the appropriate TODOs and no additional documentation is needed.

The code, tests with compilation, pycodestyle, pylint all look good. There is just some very minor inline-comments to fix and then this can be merged.

File: src/psyclone/tests/psyir/nodes/scoping_node_test.py [hunk 1]
```diff
@@ -176,8 +176,50 @@ def test_scoping_node_copy_hierarchy():
...
+    new_loop_var = new_schedule.symbol_table.lookup("idx")
+    assert new_loop_var is not loop_var
+    new_loop = new_schedule[0]
+    assert new_loop.variable is new_loop_var
+    assert new_loop.loop_body[0].rhs.symbol is new_loop_var
+    # Check that the copied tree results in compilable Fortran
+    output = fortran_writer(new_schedule)
+    assert Compile(tmpdir).string_compiles(output)
+    # Check that the copy operation succeeds, even if there is no variable
+    # assocated with the Loop (as can be the case in the LFRic domain).
```
[the remark this answers is not in the document]
Comment (arporter) [reply src/psyclone/tests/psyir/nodes/scoping_node_test.py:221]: Oops.

File: src/psyclone/tests/psyir/nodes/scoping_node_test.py [hunk 2]
```diff
@@ -38,10 +38,10 @@
 
 from __future__ import absolute_import
 from psyclone.psyir.nodes import (Schedule, Assignment, Reference, Container,
-                                  Routine, ArrayReference)
+                                  Loop, Literal, Routine, ArrayReference)
```
[the remark this answers is not in the document]
Comment (arporter) [reply src/psyclone/tests/psyir/nodes/scoping_node_test.py:41]: Oh yes.

File: src/psyclone/psyir/nodes/scoping_node.py [hunk 3]
```diff
@@ -80,10 +80,25 @@ def _refine_copy(self, other):
         self._symbol_table._node = self  # Associate to self
 
         # Update of children references to point to the equivalent symbols in
-        # the new symbol table attached to self
-        for ref in self.walk(Reference):
-            if ref.symbol in other.symbol_table.symbols:
-                ref.symbol = self.symbol_table.lookup(ref.symbol.name)
+        # the new symbol table attached to self.
```
[the remark this answers is not in the document]
Comment (arporter) [reply src/psyclone/psyir/nodes/scoping_node.py:83]: Done.

**Review by sergisiso — APPROVED** [review]
@arporter All comments addressed, this is approved for merging

**Pull request merged**